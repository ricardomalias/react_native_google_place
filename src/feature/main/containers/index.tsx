import { connect } from "react-redux";
import MainComponent from "../components";
import mainData from "../reducers";

export default connect(
    mainData,
    {

    }
)(MainComponent);


