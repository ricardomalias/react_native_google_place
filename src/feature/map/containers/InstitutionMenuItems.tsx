import {InstitutionMenuItem} from "../../../models/InstitutionMenuItem";
import {Permission} from "../../../models/Permission";


export const InstitutionMenuItems: Array<InstitutionMenuItem> = [
    // {
    //     name: "institutionCalendar",
    //     label: "institution_calendar",
    //     icon: "calendar-alt",
    //     permission: Permission.READ_INSTITUTION,
    //     route: "InstitutionCalendar"
    // },
    // {
    //     name: "lessons",
    //     label: "lesson.plural",
    //     icon: "book",
    //     permission: Permission.READ_OWN_LESSON,
    //     route: "LessonWeek"
    // },
    {
        name: "reportCard",
        label: "report_card.plural",
        icon: "graduation-cap",
        permission: Permission.READ_OWN_GRADE,
        route: "Gradetable"
    },
    {
        name: "finance",
        label: "finance",
        icon: "money-bill",
        permission: Permission.READ_OWN_DEBT,
        route: "Finance"
    },
    {
        name: "classrooms",
        label: "classroom.plural",
        icon: "chalkboard",
        permission: Permission.READ_OWN_CLASSROOM,
        route: "Classroom"
    },
]