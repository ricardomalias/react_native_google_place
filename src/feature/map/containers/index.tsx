import { connect } from "react-redux";
import MapComponent from "../components";
import MapData from "../reducers";

// import {getInstitutionMemberPermission} from "../../../store/map/actions";

export default connect(
    MapData,
    {
        // getInstitutionMemberPermission
    }
)(MapComponent);
