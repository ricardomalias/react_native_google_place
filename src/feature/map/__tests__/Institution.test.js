import React from 'react';
import renderer from 'react-test-renderer';
import Institution from "../components";
import {shallow, mount} from "enzyme";
import sinon from "sinon"
import {InstitutionMenuItems} from "../containers/InstitutionMenuItems";


describe('<Institution />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Institution />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('with shallow', () => {
        const wrapper = shallow(<Institution />);

        expect(wrapper.find('FlContainer')).toBeDefined();
        expect(wrapper.find('FlContainer')).toHaveLength(1);
    });

    // it('with mount', () => {
    //     const permissions = ["READ_OWN_GRADE"]
    //     Text.prototype.componentDidMount = jest.fn(e => e);
    //     mount(<Institution getInstitutionMemberPermission={() => {}} isLoadingData={false} permissions={permissions} />);
    //     const spy = sinon.spy(Institution.prototype, "componentDidMount");
    //     expect(spy).toHaveBeenCalledTimes(1);
    //     // const wrapper = mount(<Text>oi</Text>);
    //     // console.log(wrapper.debug())
    // });

    it('with permission has at least 1 child', () => {
        const permissions = ["READ_OWN_GRADE"]
        const wrapper = mount(<Institution getInstitutionMemberPermission={() => {}} isLoadingData={false} permissions={permissions} />);
        wrapper.setState(InstitutionMenuItems)
        wrapper.update();

        expect(wrapper.find('InstitutionMenuList')).toHaveLength(1)

        // console.log(wrapper.debug());
    });

    it('without permission has 0 child', () => {
        const permissions = []
        const wrapper = mount(<Institution getInstitutionMemberPermission={() => {}} isLoadingData={false} permissions={permissions} />);
        wrapper.setState(InstitutionMenuItems)
        wrapper.update();

        expect(wrapper.find('InstitutionMenuList')).toHaveLength(0)
    });
});