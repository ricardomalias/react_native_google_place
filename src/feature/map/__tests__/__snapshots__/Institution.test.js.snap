// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Institution /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 120,
      }
    }
    onContentSizeChange={[Function]}
    onScroll={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={16}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
